src/backend/routes/api.js

```diff
@@ -1,442 +1,562 @@
-module.exports = function(io){
-
-const express = require('express');
-const gameIo = io.of('/game');
-const router = express.Router();
-const queries = require('../db/queries.js');
-
-//retrieve games
-router.get('/games', async (req, res, next) => {
-  try{
-    const games = await queries.getGames();
-    res.send({games: games});
-  }catch(e){
-    res.send({error: e});
-  }
-});
-
-//get game
-router.get('/game/:id', async (req, res, next) => {
-  const id = parseInt(req.params['id'], 10);
-  try{
-    const game = await queries.findGame({gameId: id});
-    res.send({game: game[0]});
-  }catch(e){
-    res.send({error: e});
-  }
-});
+module.exports = function(io) {
+  const express = require("express");
+  const gameIo = io.of("/game");
+  const router = express.Router();
+  const queries = require("../db/queries.js");
+
+  //retrieve games
+  router.get("/games", async (req, res, next) => {
+    try {
+      const games = await queries.getGames();
+      res.send({ games: games });
+    } catch (e) {
+      res.send({ error: e });
```

Comment:
if your error handler handles exceptions correctly you can remove all the try/catch blocks in your routes 🙃

Reply:
This would make me very happy! When I make a call to one of the queries should I pass in the next callback? For example:
```const games = await queries.getGames(next)
```
That would give me the chance to forward the error to the error handler in app.js.

Is there a better way of doing this?

Reply:
you could do that as the express docs suggest or alternatively use something like https://github.com/express-promise-router/express-promise-router which allows you to make your routes `async` and throw errors as they normally would without passing around the next callback

Reply:
I ended up making a async handler which does something similar to what you suggested. Below is the async handler which is wrapped around each route.

`const AsyncHandler = fn => (res, req, next) => {
    Promise.resolve(fn(res, req, next)).catch(err => {
      next(err);
    })
}

module.exports = AsyncHandler;`

The queries.js file throws errors if something goes wrong which is then caught by this handler.